projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-ai-feature/index.ts

```diff
@@ -82,13 +82,32 @@ export async function getAIFeatures(): Promise< AIFeatureProps > {
 
 export default function useAIFeature() {
 	const [ data, setData ] = useState< AIFeatureProps >( AI_Assistant_Initial_State );
+	const [ loading, setLoading ] = useState< boolean >( false );
+	const [ failed, setFailed ] = useState< boolean >( false );
+
+	const loadFeatures = async () => {
+		setLoading( true );
+		setFailed( false );
+
+		try {
+			const aiFeatures = await getAIFeatures();
+			setData( aiFeatures );
```

Comment:
should set `failed` to `false` here?

Reply:
Also, does it make sense to store the error instead of the boolean state? It can be provided by the `getAIFeatures()` helper.

Reply:
> should set failed to false here?
`error` is set to `null` when the loading starts, as that is a new request, so no need to set it on success.